### PR TITLE
Add relative `uri` to MenuItem

### PR DESCRIPTION
**Pull request by josephfusco**
What does this implement/fix? Explain your changes.
---------------------------------------------------
This fixes an issue with using wp-graphql in a WordPress multisite using subdirectories where the individual site subdirectory is included with the menu path.

Does this close any currently open issues?
------------------------------------------
https://github.com/wp-graphql/wp-graphql/issues/2362

Any relevant logs, error output, GraphiQL screenshots, etc?
-------------------------------------
**Before**
Screen Shot 2022-05-04 at 3 24 03 PM

**After**
Screen Shot 2022-05-04 at 3 23 49 PM

Where has this been tested?
---------------------------
**Operating System:** macOS 12.3.1
**WordPress Version:** 5.9.2

### CODE REVIEW

**Comment by justlevine** [comment]
I think a better approach to this would be to add a `uri` field in addition to the existing `url`, instead of changing the expectation (and breaking behavior) of either `url` (a fully qualified url) and `path` (the relative path on the server).

This is also how `User` objects do it:
image

**Comment by josephfusco** [comment]
@justlevine I added a new `uri` field rather than modifying the pre-existing `path` as suggested.

Here's how it looks now

Screen Shot 2022-05-05 at 1 08 47 PM

**Comment by justlevine** [comment]
Hey @josephfusco , I tweaked your code a bit to follow the patterns used for other `uri` fields. Can you confirm it still works on multisite? (It should be fine, but multisite support is still a bit spotty ).

It seems that `MenuItems` aren't really tested, so I'm gonna take a minute to backfill those before marking this ready.

**Comment by josephfusco** [comment]
Hey @justlevine I greatly appreciate your help with this!

After giving this more thought, I'm going to close this out add to fix to the faustwp WordPress plugin. It feels like the more appropriate place to put this seeing that faustwp is the one that is replacing the subdirectory.

**Comment by justlevine** [comment]
@josephfusco I believe this should stay open. While its true that Faust is doing a bad replace on the url, there are a lot of use cases for a relative uri, and it makes the object behave more uniformly with other `Nodes` that can have an associated link. 

If you prefer, I can create a separate PR - lemme know!

**Comment by josephfusco** [comment]
@justlevine Works for me! Feel free to keep it on this PR :)

**Comment by justlevine** [comment]
@jasonbahl not sure why the schema linter is failing 🤔

**Comment by jasonbahl** [comment]
@justlevine looks probably related with this latest release of the schema linter: https://github.com/cjoudrey/graphql-schema-linter/releases/tag/v3.0.0

**Comment by jasonbahl** [comment]
looks like we'll need to explicitly install graphql in the workflow now

**Comment by jasonbahl** [comment]
I just fixed the Schema linter in #2368

**Comment by justlevine** [comment]
@jasonbahl  And it's passing again 🚀. Ready for your review :)

**Comment by jasonbahl** [comment]
Ah issue for multisite tests here: https://github.com/wp-graphql/wp-graphql/issues/2275

**Comment by jasonbahl** [comment]
Ok, so I got #2375 merged, so now we have tests running in multisite! 

I'll work up a test for this so we can get this merged.

**Comment by justlevine** [comment]
> I'll work up a test for this so we can get this merged.

Only thing that should be missing is an `assertEquals()` for `$actual['data']['menuItem']['path']`;

File: src/Type/ObjectType/MenuItem.php [hunk 1]
```diff
@@ -21,7 +21,7 @@ public static function register_type() {
 			'MenuItem',
 			[
 				'description' => __( 'Navigation menu items are the individual items assigned to a menu. These are rendered as the links in a navigation menu.', 'wp-graphql' ),
-				'interfaces'  => [ 'Node', 'DatabaseIdentifier' ],
+				'interfaces'  => [ 'Node', 'DatabaseIdentifier', 'UniformResourceIdentifiable' ],
```
Comment (jasonbahl) [inline src/Type/ObjectType/MenuItem.php:24]: Ah, so even though the `MenuItem` type has a `uri` field, Menu Items are not actually uniquely identifiable by a URI. 

If you take the `MenuItem.uri` field and do a `nodeByUri( $uri: MenuItem.uri ) { ... }` query, you will get another node (Post, Page, etc)

The `MenuItem` Type should not implement the `UniformResourceIdentifiable` interface, as a MenuItem is not identifiable by a URI. It links to other nodes that are identifiable by uri, but it by itself is not.

Comment (jasonbahl) [reply src/Type/ObjectType/MenuItem.php:24]: If we look in the Schema now for the `nodeByUri` field, we see that it returns `UniformResourceIdentifiable` Interface, which now has `MenuItem` as a Possible Type, which is misleading, as `nodeByUri` will never return an object with type `MenuItem`

CleanShot 2022-05-13 at 16 32 49

**Comment by jasonbahl** [comment]
I left some notes about the `uri` field / `UniformResourceIdentifiable` Interface. 

I think I'm ok with having the `uri` field as a non-breaking change here, but I think we should remove the `UniformResourceIdentifiable` interface, as this Type is not actually identifiable by a uri. The nodes that MenuItem links to are identifiable by the uri, but not the MenuItem itself.

**Comment by jasonbahl** [comment]
I pushed up a commit to remove `UniformResourceIdentifiable` from the `MenuItem` type

**Comment by jasonbahl** [comment]
Pushed a follow-up commit adding the `uri` field back to the `MenuItem` Type since it was removed when removing the `UniformResourceIdentifiable` Interface from the Type. 

Now, we can query the `uri` field of a MenuItem, but `MenuItem` will not be shown as a `possibleType` in `nodByUri` queries.

**Comment by jasonbahl** [comment]
I think we still might need to have a specific test where we do something like the following:

- create a new blog in the multisite
- switch to blog
- create a menu with some menu items 
- assign the menu to a location (so it's public)
- query the menu
- assert that the blog directory is _not_ in the uri

🤔

**Pull request merged**